### PR TITLE
Prevent navigation on url input suggestion selection via enter key

### LINKED ISSUE #38950: Media Replace opens new media when changing the URI of the old media
### Description

The media replace dropdown has a URL indicator and an edit button. When the edit button is pressed the URL can be edited via an input. When pressing Enter on the keyboard the URL is used to replace the media. But in Firefox it also opens the media.

### Step-by-step reproduction instructions

1. Create a new post
2. Insert an image block
3. Add an image from your media library
4. Copy this URL: `https://live.staticflickr.com/3784/12091853594_b6f2668721_b.jpg`
5. In your image block click on "Replace" in the block's toolbar
6. From the menu click on the pencil icon
7. Select the current media URL in the input that appears (tiny!)
8. Paste the URL you copied from above
9. Press the Enter key on the keyboard
10. Notice bug

### Screenshots, screen recording, code snippet

Uploading replace-URL.mp4…

### Environment info

Wordpress tunk
Gutenberg trunk
Firefox latest

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### LINKED ISSUE #40449: Button block opens new tab after setting link with enter key in Firefox
### Description

After typing or selecting a link for a button, hitting the enter key opens that link right away. It should simply save the link and allow you to continue editing.

### Step-by-step reproduction instructions

1. Insert Buttons block
2. Click the link icon for the button
3. Type out a URL
4. Hit enter key

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/162595/164058905-2544765f-632d-4a36-8f6b-6c4538af3457.mp4

### Environment info

- WordPress 5.9.3
- Firefox 99.0
- Ubuntu 20.04

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by draganescu**
## What?
Fixes #38950
Fixes #40449

## Why?
We don't have to navigate to selected urls.

## How?
For the enter key on the suggestion list for url input there is a select behavior which did not have preventDefault called on the key event.

## Testing Instructions
1. Create a new post
2. Insert an image block
3. Add an image from your media library
4. Copy this URL: `https://live.staticflickr.com/3784/12091853594_b6f2668721_b.jpg`
5. In your image block click on "Replace" in the block's toolbar
6. From the menu click on the pencil icon
7. Select the current media URL in the input that appears (tiny!)
8. Paste the URL you copied from above
9. Press the Enter key on the keyboard
10. The page should not navigate away and the image is changed.

## Other

The changes seem to come from https://github.com/WordPress/gutenberg/pull/32552 but not 100% sure and also there is a comment that says

> // We shouldn't preventDefault to allow block arrow keys navigation.

... but I don't think this is about the ENTER key. In the past (looking through commits the enter key was handled directly in the button's onClick prop. Later it had a stopPropagation, not it has no handling, hence the error.

cc @youknowriad for a trip down memory lane if there is anything from the past refactoring that required the ENTER key to not be handled.

### CODE REVIEW

**Comment by alexstine** [comment]
@draganescu I'll check this out later today and make sure it still meets A11Y requirements. Just a side note, I added "Accessibility" label so that way it could be more easily found. There are so many sub accessibility labels it would take me ages to look through them all. Generally, I always tag with "Accessibility" and whatever necessary sub-label.

Thanks.

**Review by Mamaduka — APPROVED** [review]
The fix works as expected ✅  Tested in Chrome and Firefox.

I'm approving the PR, but let's wait for @alexstine's a11y review.

**Comment by alexstine** [comment]
> From the menu click on the pencil icon

Which menu option is that?
- Open Media Library
- Upload

Thanks.

**Comment by Mamaduka** [comment]
@alexstine

That would be the "Edit" button in the "Replace" dropdown.

**Comment by alexstine** [comment]
@Mamaduka I do not see an Edit option in the Replace dropdown. Just the options I posted in my last comment. 😕

**Comment by Mamaduka** [comment]
@alexstine, It's usually the last option in that dropdown — part of the "Current media URL:" form.

P.S. We probably should make that text a proper label.

**Comment by draganescu** [comment]
@alexstine to get to the pencil icon with the keyboard once you are in the replace drop down menu you use arrow keys to navigate the 1st two options *but* you need to press *tab* to get to the next section of the drop down menu where the edit button is. Pressing tab will focus the edit button.

**Review by alexstine — APPROVED** [review]
LGTM. 👍

**Comment by alexstine** [comment]
@draganescu Could you please create a follow-up issue to change how this works? Pressing Tab to get to the Edit button is horribly confusing and there is no way to navigate back. This block needs help. This is an example of how you can create something so inaccessible that I didn't even know about it unless told. It presents a real problem as I would have never found this during testing. 😞 

Thanks.

**Comment by TimothyBJacobs** [comment]
This looks to be a breaking change. If you were using a `URLInput` inside of a `<form>`, a user hitting enter will no longer trigger the wrapping form submission. You instead need to populate the `onSubmit` prop on `URLInput` which isn't documented.

I'm not sure if there is a good way to fix the BC issue without also reverting the fix. But I wanted to flag it as it broke one of our blocks when updating to 6.1.

Example code:

```js
<form
	onSubmit={ ( e ) => {
		e.preventDefault();
		setAttributes( {
			productHref: editingLink,
		} );
		onClose();
	} }
>
	<URLInput
		disableSuggestions
		value={ editingLink }
		onChange={ setEditingLink }
	/>
</form>
```

**Comment by draganescu** [comment]
@TimothyBJacobs oups! How should we document it best for those affected?

**Comment by TimothyBJacobs** [comment]
I think a note in the changelog that points to `onSubmit` would be good. I'm not sure it's a major enough change to warrant mentioning in the [component changes dev note](https://make.wordpress.org/core/2022/10/10/editor-components-updates-for-wordpress-6-1/).

**Comment by Mamaduka** [comment]
After #50158 is merged, this should be less of a breaking change.

**Pull request merged**
